checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+   * @return true iff we should transfer ownership to the return type of the method corresponding to
+   *     a CFG
+   */
+  private boolean isTransferOwnershipAtReturn(ControlFlowGraph cfg) {
+    if (checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)) {
+      // If not using LO, default to always transferring at return, just like ECJ does.
+      return true;
+    }
+
+    UnderlyingAST underlyingAST = cfg.getUnderlyingAST();
+    if (underlyingAST instanceof UnderlyingAST.CFGMethod) {
+      // TODO: lambdas? I think in that case we'll return false, below.
+      MethodTree method = ((UnderlyingAST.CFGMethod) underlyingAST).getMethod();
+      ExecutableElement executableElement = TreeUtils.elementFromDeclaration(method);
+      return typeFactory.getDeclAnnotation(executableElement, NotOwning.class) == null;
+    }
+    return false;
+  }
+
+  /**
+   * Updates a set of obligations to account for an assignment. Assigning to an owning field might
+   * remove obligations, assigning to a new local variable might modify an obligation (by increasing
+   * the size of its resource alias set), etc.
+   *
+   * @param node the assignment
+   * @param obligations the set of obligations to update
+   */
+  private void handleAssignment(
+      AssignmentNode node, Set<ImmutableSet<LocalVarWithTree>> obligations) {
+    Node lhs = node.getTarget();
+    Element lhsElement = TreeUtils.elementFromTree(lhs.getTree());
+    // Use the temporary variable for the rhs if it exists.
+    Node rhs = removeCasts(node.getExpression());
+    LocalVariableNode tempVarForRhs = typeFactory.getTempVarForNode(rhs);
+    if (tempVarForRhs != null) {
+      rhs = tempVarForRhs;
+    }
+
+    // Ownership transfer to @Owning field.
+    if (lhsElement.getKind() == ElementKind.FIELD) {
+      boolean isOwningField =
+          !checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)
+              && typeFactory.getDeclAnnotation(lhsElement, Owning.class) != null;
+      // Check that there is no obligation on the lhs, if the field is non-final and owning.
+      if (isOwningField
+          && typeFactory.canCreateObligations()
+          && !ElementUtils.isFinal(lhsElement)) {
+        checkReassignmentToField(node, obligations);
+      }
+      // Remove obligations from local variables, now that the owning field is responsible.
+      // (When obligation creation is turned off, non-final fields cannot take ownership.)
+      if (isOwningField
+          && rhs instanceof LocalVariableNode
+          && (typeFactory.canCreateObligations() || ElementUtils.isFinal(lhsElement))) {
+        removeObligationContainingVar(obligations, (LocalVariableNode) rhs);
+      }
+    } else if (lhs instanceof LocalVariableNode) {
+      LocalVariableNode lhsVar = (LocalVariableNode) lhs;
+      doGenKillForPseudoAssignment(node, obligations, lhsVar, rhs);
+    }
+  }
+
+  /**
+   * Remove any obligations that contain {@code var} in their resource-alias set.
```

Comment:
I find this wording a bit confusing because an obligation *is* a resource-alias set; the variable `setContainingRhs` has the same type as `obligation`.

Reply:
Does this same comment also apply to the method `getResourceAliasSetForVar`? It also returns an obligation (which is also a resource alias set), but IMO in most of its use cases `getResourceAliasSetForVar` is more useful for understanding what it does - this code is the exception.